python-checks/src/main/java/org/sonar/python/checks/BooleanCheckNotInvertedCheck.java

```diff
@@ -120,8 +122,36 @@ private static void createQuickFix(IssueWithQuickFix issue, String oppositeOpera
   }
 
   private static PythonTextEdit getReplaceEdit(BinaryExpression toUse, Expression toReplace, String oppositeOperator) {
-    List<Tree> children = toReplace.parent().parent().children();
+    // To be able to remove the 'not' sign and the possible space afterwards
+    Tree notAncestor = Optional.ofNullable(TreeUtils.firstAncestorOfKind(toReplace, Kind.NOT))
+      .orElseThrow(() -> new IllegalArgumentException("A NOT tree must be found. It was not the case."));
+
+    List<Tree> children = notAncestor.children();
     return PythonTextEdit.replaceRange(children.get(0), children.get(children.size() - 1),
-      toUse.leftOperand().firstToken().value() + " " + oppositeOperator + " " + toUse.rightOperand().firstToken().value());
+      getNewExpression(toUse, oppositeOperator));
+  }
+
+  private static String getNewExpression(BinaryExpression toUse, String oppositeOperator) {
+    return getText(toUse.leftOperand()) + " " + oppositeOperator + " " + getText(toUse.rightOperand());
+  }
+
+  private static String getText(Tree tree) {
+    if (tree.children().isEmpty()) {
+      if (tree.is(Kind.EXPRESSION_LIST)) {
+        return "";
+      }
+      return ((Token) tree).value();
+    }
+    // Regular argument and tuple may have children
+    if (tree.children().size() == 1 && !tree.children().get(0).is(Kind.REGULAR_ARGUMENT, Kind.TUPLE, Kind.PARENTHESIZED)) {
+      return tree.firstToken().value();
+    }
+    return tree.children().stream()
+      .map(BooleanCheckNotInvertedCheck::getText)
+      .reduce("", (acc, b) -> isSpaceNotNeeded(acc, b) ? (acc + b) : (acc + " " + b));
```

Comment:
why `b`? I would maybe name it `currentChar` or `toAdd`

Reply:
Yep, it would be clearer